src/systems/cursor-targetting-system.js

```diff
@@ -0,0 +1,44 @@
+export class CursorTargettingSystem {
+  constructor() {
+    this.targets = [];
+    this.setDirty = this.setDirty.bind(this);
+    this.dirty = true;
+    // TODO: Use the MutationRecords passed into the callback function to determine added/removed nodes!
+    this.observer = new MutationObserver(this.setDirty);
+    const scene = document.querySelector("a-scene");
+    this.observer.observe(scene, { childList: true, attributes: true, subtree: true });
+    scene.addEventListener("object3dset", this.setDirty);
+    scene.addEventListener("object3dremove", this.setDirty);
+    this.rightRemote = document.querySelector("#cursor-controller");
+  }
+
+  setDirty() {
+    this.dirty = true;
+  }
+
+  tick(t) {
+    if (this.dirty) {
+      this.populateEntities(this.targets);
+      this.dirty = false;
+    }
+
+    this.rightRemote.components["cursor-controller"].tick2(t);
+  }
+
+  populateEntities(targets) {
+    targets.length = 0;
+    // TODO: Do not querySelectorAll on the entire scene every time anything changes!
```

Comment:
This isn't new, this was just taken from cursor controller... Still, this is slow and it happens all the time!!